k8s/tools/k8s_ssh_security.sh

```diff
@@ -0,0 +1,157 @@
+
+    NODES_QUERY=".SecurityGroups[] | select(.GroupName == \"${NODES_GROUP_NAME}\") | .GroupId"
+    NODES_GROUP_ID=$(aws ec2 describe-security-groups --region ${KOPS_REGION} | jq -r "${NODES_QUERY}")
+    echo "Masters group: ${MASTERS_GROUP_NAME} : ${MASTERS_GROUP_ID}"
+    echo "Nodes group: ${NODES_GROUP_NAME} : ${NODES_GROUP_ID}"
+}
+
+
+show_current_security() {
+    get_security_groups
+    QUERY=".SecurityGroups[0].IpPermissions[] | {ip_protocol: .IpProtocol, from: .FromPort, to: .ToPort, cidr: .IpRanges[0].CidrIp, groups: [.UserIdGroupPairs[].GroupId]}"
+    echo "Master security groups:"
+    aws ec2 describe-security-groups --group-id ${MASTERS_GROUP_ID} --region ${KOPS_REGION} | jq -c -r "${QUERY}"
+    echo ""
+    echo "Nodes security groups:"
+    aws ec2 describe-security-groups --group-id ${NODES_GROUP_ID} --region ${KOPS_REGION} | jq -c "${QUERY}"
+}
+
+get_my_ip() {
+    which dig > /dev/null
+    if [ $? -ne 0 ]; then
+        echo "This script needs the dig command installed"
+        echo "Try this: sudo apt-get install dnsutils"
+        exit 1
+    fi
+
+    MY_PUBLIC_IP=$(dig +short myip.opendns.com @resolver1.opendns.com.)
+}
+
+enable_ssh() {
+    get_security_groups
+    if [ ! -z "${MYIP}" ]; then
+        get_my_ip
+        this_cidr="${MY_PUBLIC_IP}/32"
+        echo "Limiting SSH to ${thisip}"
+        aws ec2 authorize-security-group-ingress --group-id ${MASTERS_GROUP_ID} --port 22 --protocol tcp --cidr ${this_cidr} --region ${KOPS_REGION}
+        aws ec2 authorize-security-group-ingress --group-id ${NODES_GROUP_ID} --port 22 --protocol tcp --cidr ${this_cidr} --region ${KOPS_REGION}
+    else
+        echo "SSH open to 0.0.0.0/0"
+        aws ec2 authorize-security-group-ingress --group-id ${MASTERS_GROUP_ID} --port 22 --protocol tcp --cidr 0.0.0.0/0 --region ${KOPS_REGION}
+        aws ec2 authorize-security-group-ingress --group-id ${NODES_GROUP_ID} --port 22 --protocol tcp --cidr 0.0.0.0/0 --region ${KOPS_REGION}
+    fi
+    echo "DONE"
+}
+
+disable_ssh() {
+    get_security_groups
+    echo "Disabling SSH"
+    if [ ! -z "${MYIP}" ]; then
+        get_my_ip
+        this_cidr="${MY_PUBLIC_IP}/32"
+        aws ec2 revoke-security-group-ingress --group-id ${MASTERS_GROUP_ID} --port 22 --protocol tcp --cidr ${this_cidr} --region ${KOPS_REGION}
+        aws ec2 revoke-security-group-ingress --group-id ${NODES_GROUP_ID} --port 22 --protocol tcp --cidr ${this_cidr} --region ${KOPS_REGION}
+    else
+        aws ec2 revoke-security-group-ingress --group-id ${MASTERS_GROUP_ID} --port 22 --protocol tcp --cidr 0.0.0.0/0 --region ${KOPS_REGION}
+        aws ec2 revoke-security-group-ingress --group-id ${NODES_GROUP_ID} --port 22 --protocol tcp --cidr 0.0.0.0/0 --region ${KOPS_REGION}
+    fi
+    echo "DONE"
+}
+
+for arg in "$@"
+do
+    case $arg in
+        --show)
+        SHOW=1
+        shift
+        ;;
+        --myip)
```

Comment:
This is great, but as we discussed yesterday I'd like us to also accept an IP directly. Not a blocker, a TODO